From f911ddbc7db1ac864278f6c07ccc72ecbf7e5d2e Mon Sep 17 00:00:00 2001
From: Dinolek <git@dinolek.me>
Date: Tue, 8 Jul 2025 01:06:06 +0200
Subject: [PATCH 064/115] arm64: dts: mediatek: mt6785: Add usb support

Add ssusb, xhci and u3phy nodes needed for usb support.

Signed-off-by: Dinolek <git@dinolek.me>
---
 arch/arm64/boot/dts/mediatek/mt6785.dtsi | 49 ++++++++++++++++++++++++
 1 file changed, 49 insertions(+)

diff --git a/arch/arm64/boot/dts/mediatek/mt6785.dtsi b/arch/arm64/boot/dts/mediatek/mt6785.dtsi
index bb6fa67b3..11f33edfd 100644
--- a/arch/arm64/boot/dts/mediatek/mt6785.dtsi
+++ b/arch/arm64/boot/dts/mediatek/mt6785.dtsi
@@ -8,6 +8,7 @@
 #include <dt-bindings/interrupt-controller/arm-gic.h>
 #include <dt-bindings/interrupt-controller/irq.h>
 #include <dt-bindings/memory/mediatek,mt6785-memory-port.h>
+#include <dt-bindings/phy/phy.h>
 #include <dt-bindings/pinctrl/mediatek,mt6785-pinfunc.h>
 #include <dt-bindings/power/mediatek,mt6785-power.h>
 #include <dt-bindings/reset/mediatek,mt6785-resets.h>
@@ -788,6 +789,33 @@ spi7: spi@1101e000 {
 			status = "disabled";
 		};
 
+		ssusb: usb@11201000 {
+			compatible = "mediatek,mt6785-mtu3", "mediatek,mtu3";
+			reg = <0 0x11201000 0 0x2e00>, <0 0x11203e00 0 0x0100>;
+			reg-names = "mac", "ippc";
+			ranges = <0 0 0 0x11200000 0 0x3f00>;
+			#address-cells = <2>;
+			#size-cells = <2>;
+			interrupts = <GIC_SPI 96 IRQ_TYPE_LEVEL_LOW>;
+			clocks = <&infracfg CLK_INFRA_SSUSB>,
+				 <&infracfg CLK_INFRA_SSUSB_XHCI>;
+			clock-names = "sys_ck", "ref_ck";
+			phys = <&u2port PHY_TYPE_USB2>, <&u3port PHY_TYPE_USB3>;
+			wakeup-source;
+			mediatek,syscon-wakeup = <&pericfg 0x420 102>;
+			status = "disabled";
+
+			xhci: usb@0 {
+				compatible = "mediatek,mt6785-xhci", "mediatek,mtk-xhci";
+				reg = <0 0 0 0x1000>;
+				reg-names = "mac";
+				interrupts = <GIC_SPI 97 IRQ_TYPE_LEVEL_LOW>;
+				clocks = <&infracfg CLK_INFRA_SSUSB_XHCI>;
+				clock-names = "sys_ck";
+				status = "disabled";
+			};
+		};
+
 		audsys: clock-controller@11210000 {
 			compatible = "mediatek,mt6785-audsys", "syscon";
 			reg = <0 0x11210000 0 0x2000>;
@@ -817,6 +845,27 @@ ufshci: ufshci@11270000 {
 			status = "disabled";
 		};
 
+		u3phy: t-phy@11e40000 {
+			compatible = "mediatek,mt6785-tphy", "mediatek,generic-tphy-v2";
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges = <0x0 0x0 0x11e40000 0x1000>;
+
+			u2port: usb-phy@0 {
+				reg = <0x0 0x700>;
+				clocks = <&clk26m>;
+				clock-names = "ref";
+				#phy-cells = <1>;
+			};
+
+			u3port: usb-phy@700 {
+				reg = <0x700 0x900>;
+				clocks = <&clk26m>;
+				clock-names = "ref";
+				#phy-cells = <1>;
+			};
+		};
+
 		mipi_tx0: dsi-phy@11e50000 {
 			compatible = "mediatek,mt8183-mipi-tx";
 			reg = <0 0x11e50000 0 0x1000>;
-- 
2.56.0

